test/03_device-state.ts

```diff
@@ -38,296 +38,315 @@ configMock['API_HEARTBEAT_STATE_TIMEOUT_SECONDS'] = TIMEOUT_SEC;
 // register the mocks...
 mockery.registerMock('../src/lib/config', configMock);
 
-describe('Device State v2', () => {
-	let fx: fixtures.Fixtures;
-	let admin: UserObjectParam;
-	let applicationId: number;
-	let device: fakeDevice.Device;
-
-	const tracker = new StateTracker();
-	const updateDeviceModel = stateMock.getInstance()['updateDeviceModel'];
+['v2', 'v3'].forEach((stateVersion) =>
+	describe(`Device State ${stateVersion}`, () => {
+		let fx: fixtures.Fixtures;
+		let admin: UserObjectParam;
+		let applicationId: number;
+		let device: fakeDevice.Device;
+
+		const tracker = new StateTracker();
+		const updateDeviceModel = stateMock.getInstance()['updateDeviceModel'];
+
+		before(async () => {
+			fx = await fixtures.load('03-device-state');
+
+			admin = fx.users.admin;
+			applicationId = fx.applications.app1.id;
+
+			// create a new device in this test application...
+			device = await fakeDevice.provisionDevice(admin, applicationId);
+
+			stateMock.getInstance()['updateDeviceModel'] = function (
+				uuid: string,
+				newState: stateMock.DeviceOnlineStates,
+			) {
+				tracker.stateUpdated(uuid, newState);
+				return updateDeviceModel.call(this, uuid, newState);
+			};
 
-	before(async () => {
-		fx = await fixtures.load('03-device-state');
+			mockery.registerMock('../src/lib/device-online-state', stateMock);
+		});
 
-		admin = fx.users.admin;
-		applicationId = fx.applications.app1.id;
+		after(async () => {
+			await fixtures.clean(fx);
+			mockery.deregisterMock('../src/lib/env-vars');
+			mockery.deregisterMock('../src/lib/config');
+			mockery.deregisterMock('../src/lib/device-online-state');
+		});
 
-		// create a new device in this test application...
-		device = await fakeDevice.provisionDevice(admin, applicationId);
-
-		stateMock.getInstance()['updateDeviceModel'] = function (
-			uuid: string,
-			newState: stateMock.DeviceOnlineStates,
-		) {
-			tracker.stateUpdated(uuid, newState);
-			return updateDeviceModel.call(this, uuid, newState);
-		};
+		describe(`API heartbeat state`, () => {
```

Comment:
This suite doesn't call the target state endpoint, we could take it out but it would require some code duplication